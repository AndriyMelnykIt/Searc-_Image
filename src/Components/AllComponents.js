export { default as HomePage } from './HomePage';
export { default as PageNotFound } from './PageNotFound';
export { default as Picture } from './Picture';
export { default as Header } from './Header';
export { default as Photos } from './Photos';
export { default as Spinner } from './Spinner';
export { default as Pages } from './Pages';
export { default as SearchPage } from './SearchPage';
export { default as SearchHistory } from './SearchHistory';
export { default as PhotosPage } from './PhotosPage';

