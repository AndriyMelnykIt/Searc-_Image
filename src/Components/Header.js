import React from 'react';
import {Navbar, Container, FormControl} from "react-bootstrap";

const Header = () => {
    return (
        <div>
            <Navbar bg='dark' variant='dark' expand='lg' collapseOnSelect>
                <Container fluid>
                    <FormControl
                        placeholder="Поле пошуку"
                        aria-label="Username"
                    />
                </Container>
            </Navbar>
        </div>
    );
};

export default Header;