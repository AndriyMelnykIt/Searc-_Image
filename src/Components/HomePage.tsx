import React from 'react';

import { SearchHistory, SearchPage } from './index';

const HomePage = () => (
    <>
        <SearchPage/>
        <SearchHistory/>
    </>
);

export default HomePage;