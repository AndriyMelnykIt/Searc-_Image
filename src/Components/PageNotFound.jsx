import React from 'react';
import { StylePageNotFound } from '../styled/PageNotFound';
import { NavLink } from 'react-router-dom';

const PageNotFound = () => (
        <StylePageNotFound>
           <div>404 Error Page</div>
            <div>Sorry, this page does't exist</div>
            <NavLink to='/'> Go Back </NavLink>
        </StylePageNotFound>
    );
export default PageNotFound;