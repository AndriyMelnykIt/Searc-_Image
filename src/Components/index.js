export { default as Homepage } from './Homepage';
export { default as PageNotFound } from './PageNotFound';
export { default as Picture } from './Picture';
