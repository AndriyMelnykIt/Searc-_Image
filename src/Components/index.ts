export { default as HomePage } from './HomePage';
export { default as Picture } from './WaltyPhotos/Picture';
export { default as Header } from './Header';
export { default as Photos } from './WaltyPhotos/Photos';
export { default as Spinner } from './Spinner';
export { default as Pages } from './Pages';
export { default as SearchPage } from './WaltyPhotos/SearchPage';
export { default as SearchHistory } from './WaltyPhotos/SearchHistory';
