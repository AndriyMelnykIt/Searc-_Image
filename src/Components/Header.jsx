import React from 'react';
import {Nav, Navbar} from 'react-bootstrap';
import {Link} from 'react-router-dom';
import {StyledHeader} from '../styled/Header';

const Header = () => {
    return (
        <>
            <StyledHeader>
                <Navbar bg='dark' variant='dark' expand='lg' collapseOnSelect>
                    <Navbar.Brand>
                        Walty Photos
                    </Navbar.Brand>
                    <Nav>
                        <Nav.Link><Link to='/'>Home</Link></Nav.Link>
                        <Nav.Link><Link to='/photo'>Photo</Link></Nav.Link>
                    </Nav>
                </Navbar>
            </StyledHeader>
        </>
    );
};

export default Header;

