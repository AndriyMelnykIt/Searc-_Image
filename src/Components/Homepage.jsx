import React, { useState, useEffect } from 'react';
import { Row, Col, Container, FormControl, Navbar } from 'react-bootstrap';

import Picture from './Picture';

import { getPhotosData } from '../api/api';
import Header from './Header';

const Homepage = () => {

    const [photos, setPhotos] = useState([]);
    const [search, setSearch] = useState('');


    const getPhotosList = async () => {
        let photosArray = [];
        for (let i = 1; i <= 20; i++) {
            photosArray.push(await getPhotosData(i))
        }
        setPhotos(photosArray)
    }

    useEffect(() => { getPhotosList() }, [])

    return (
        <>

                <Container fluid>
                    <Header/>
                    <FormControl
                        placeholder='Поле пошуку'
                        aria-label='Username'
                        onChange={ (event)=> {
                            setSearch(event.target.value)
                        }}
                    />
                </Container>
            <Row>
                { photos.filter( result => {
                    if (search === "") {
                        return result;
                    } else if (result.data.name.toLowerCase().includes(search.toLowerCase())) {
                        return result;
                    }
                    return false;
                }).map(p => (
                    <Col key={ p.data.name } xs={12} sm={12} md={6} lg={4} xl={3}>
                        <Picture photos={ p.data }/>
                    </Col>
                )) }
            </Row>
        </>
    );
};

export default Homepage;