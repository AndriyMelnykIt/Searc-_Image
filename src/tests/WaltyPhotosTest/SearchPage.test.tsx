import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { SearchPage } from '../../Components';

configure({adapter: new Adapter()});

const searchPhotos = 'test';

const handleClick = jest.fn();

const handleSearch = jest.fn();

let wrapper: string;

describe('SearchPage', () => {

    beforeAll(() => {
        wrapper = shallow(<SearchPage>
            searchPhotos={searchPhotos}
            onChangePhotos = {() => handleClick('testAfterHandler')}
            onSearchPhotos = {handleSearch}</SearchPage>);
    });

    it('should take a snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

});
