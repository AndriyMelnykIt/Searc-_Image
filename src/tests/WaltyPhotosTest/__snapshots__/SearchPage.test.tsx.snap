// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage should take a snapshot 1`] = `ShallowWrapper {}`;
